src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -1,248 +1,429 @@
-import UploadFileIcon from '@mui/icons-material/UploadFile';
-import FilePresentIcon from '@mui/icons-material/FilePresent';
-import { type Accept, type DropzoneOptions, useDropzone } from 'react-dropzone';
 import {
-	Box,
+	type Accept,
+	type DropzoneOptions,
+	useDropzone,
+	ErrorCode,
+} from 'react-dropzone';
+import {
 	Typography,
-	CircularProgress,
-	Avatar,
 	InputLabel,
-	List,
-	ListItem,
-	ListItemText,
+	Stack,
+	type SxProps,
+	Link,
+	LinearProgress,
 } from '@mui/material';
 import type { WidgetProps } from '@rjsf/utils';
-import { useCallback, useState } from 'react';
+import { useCallback, useMemo, useState } from 'react';
+import { IconButtonWithTracking, designTokens } from '../../..';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+	IconDefinition,
+	faCircleExclamation,
+	faFile,
+	faFileImage,
+	faTrash,
+	faUpload,
+} from '@fortawesome/free-solid-svg-icons';
+import uniq from 'lodash/uniq';
+
+const restingStyle: SxProps = {
+	borderColor: designTokens.color.border.value,
+	borderWidth: '2px',
+	backgroundColor: designTokens.color.bg.value,
+};
+
+const disabledStyle: SxProps = {
+	borderColor: designTokens.color.border.strong.value,
+	borderWidth: '2px',
+	backgroundColor: designTokens.color.bg.value,
+	opacity: '40%',
+};
+
+const dragStyle: SxProps = {
+	borderColor: designTokens.color.border.accent.strong.value,
+	borderWidth: '3px',
+	backgroundColor: designTokens.color.bg.accent.value,
+};
+
+// These are just the ones we use in the dashboard at the moment
+// (excluding XML which is a pro icon, for which we just default to faFile)
+// we can add more as we get the usecases
+const getFileTypeIcon = (fileType: string) => {
+	switch (fileType) {
+		case 'image/png':
+		case 'image/jpg':
+		case 'image/jpeg':
+			return faFileImage;
+		default:
+			return faFile;
+	}
+};
+
+const fileSizeToReadable = (size: number) => {
+	const i = Math.floor(Math.log(size) / Math.log(1000));
```

Comment:
Shouldn't it be `1024`? 🤔

Reply:
1024 is binary, I'm pretty sure react dropzone uses decimal

Reply:
I also think it should be 1024. This is still decimal, just breaking up on 1024bytes at a time